_posts/2024-12-11-security-audit.md

```diff
@@ -0,0 +1,68 @@
+
+1. The RubyGems.org Ruby on Rails application.
+2. The infrastructure and configuration hosting the site and gems.
+
+These areas were chosen because they represent the highest potential for security risks.
+
+Trail of Bits leveraged advanced tools like Semgrep for static analysis, Burp Suite Professional for dynamic testing, and Ruzzy for fuzzing critical components, ensuring a thorough and multi-layered audit. 
+
+The audit identified 33 issues, including seven medium-severity items and one high-severity item. **Notably, most of these findings do not constitute actual security breaches. Our team has been addressing each finding and using these insights to bolster RubyGems.org’s security posture.**
+
+## RubyGems.org Rails app
+
+Trail of Bits reviewed the Rails application code extensively and found that, in general, the site is built securely. Many of the issues highlighted were minor adjustments, often related to default Rails configurations that could be made more secure.
+
+For example, one notable finding involved a vulnerability in the email system, where fallback to unencrypted transmission could expose sensitive emails. The recommended fix was straightforward—replace enable_starttls_auto with enable_starttls to enforce strict TLS encryption.
+
+**Overall, the audit gave us confidence in the security of the RubyGems.org app, along with actionable recommendations for further strengthening it.**
+
+## RubyGems.org infrastructure
+
+On the infrastructure side, the audit revealed areas where RubyGems.org could align more closely with industry best practices, particularly regarding AWS configuration and access control. **While no immediate threats were identified, the findings emphasized the importance of proactive improvements.**
+
+**Findings included:**
+
+- Overly permissive AWS IAM permissions.
+- Insufficient role separation and domain isolation.
+- A hybrid infrastructure management approach, mixing manual changes with infrastructure-as-code.
+
+These gaps could create vulnerabilities if left unaddressed, but Trail of Bits provided specific recommendations for us to mitigate risks. For example, they highlighted the need for better isolation between AWS accounts and more granular access control policies. These findings have already informed a project proposal we’re developing to modernize our infrastructure and automate resource management through tools like Terraform.
+
+One of the most valuable outcomes on the infrastructure side was the clarity it brought in helping us better prioritize our efforts next year. Our team now has a roadmap for what improvements we need to make, how to implement them, and the expected impact on security.
+
+# Next steps for RubyGems.org security
+
+**We are using the audit’s findings to shape the next steps for security updates on RubyGems.org:**
+
+**- Short-term:** We’ve already implemented many small, immediate improvements, such as stricter TLS enforcement and better access controls.
```

Comment:
I think this is how the list would render correctly, and similarly with the other bullet points.

```suggestion
- **Short-term:** We’ve already implemented many small, immediate improvements, such as stricter TLS enforcement and better access controls.
```